### PR TITLE
first impl of mp4-springboot

### PR DESCRIPTION
**Pull request by yzx2107**
Hi Domi,

I'm done the impl for the mp4-springboot with all expected features and added some additional features as well:

- support configuration the supported media types
- support manaully reload medias after files changed
- support to run as standalone java app

### CODE REVIEW

File: .settings/org.eclipse.wst.common.component [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-        <property name="context-root" value="mp4"/>
-        <property name="java-output-path" value="/mp4/target/classes"/>
-    </wb-module>
-</project-modules>
+<?xml version="1.0" encoding="UTF-8"?><project-modules id="moduleCoreId" project-version="1.5.0">
+    <wb-module deploy-name="mp4">
+        <wb-resource deploy-path="/" source-path="/target/m2e-wtp/web-resources"/>
+        <wb-resource deploy-path="/WEB-INF/classes" source-path="/src/main/java"/>
+        <wb-resource deploy-path="/WEB-INF/classes" source-path="/src/main/resources"/>
+        <wb-resource deploy-path="/" source-path="/src/main/webapp"/>
```
Comment (domiyang) [inline .settings/org.eclipse.wst.common.component]: this /src/main/webapp path seems not required any more, can be deleted.

File: pom.xml [hunk 2]
```diff
@@ -1,27 +1,50 @@
...
+	<artifactId>mp4</artifactId>
+	<packaging>jar</packaging>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>mp4</name>
+
+	<parent>
+		<groupId>org.springframework.boot</groupId>
+		<artifactId>spring-boot-starter-parent</artifactId>
+		<version>2.0.4.RELEASE</version>
+		<relativePath /> <!-- lookup parent from repository -->
```
Comment (domiyang) [inline pom.xml]: the comment better to delete or move to head of this parent block.

File: src/main/java/mp4/Mp4Controller.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+		return "reloaded";
+	}
+
+	/**
+	 * Get the file list json string for all videos.
+	 * @return
+	 */
+	@GetMapping("/medias")
+	String getPlayListJson() {
+		return processVideoList().get("playListJson");
```
Comment (domiyang) [inline src/main/java/mp4/Mp4Controller.java]: the key playListJson can be extract as constant.

File: src/main/java/mp4/Mp4Controller.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	}
+
+	/**
+	 * Get the video link menu html content.
+	 * @return
+	 */
+	@GetMapping("/medias/links")
+	String getListOfFilesLink() {
+		return processVideoList().get("videoLinkMenu");
```
Comment (domiyang) [inline src/main/java/mp4/Mp4Controller.java]: the key videoLinkMenu can be extract as constant.

File: src/main/java/mp4/Mp4Controller.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+			// append the , for 2nd+ entries
+			if (!sbf.toString().endsWith("[")) {
+				sbf.append(",");
+			}
+			sbf.append("{\"url\":\"" + url + "\",\"fileName\":\"" + fileName + "\"}");
+		}
+		sbf.append("]}");
+
+		vMap.put("playListJson", sbf.toString());
+		vMap.put("videoLinkMenu", sbfMenu.toString());
```
Comment (domiyang) [inline src/main/java/mp4/Mp4Controller.java]: the key videoLinkMenu can be extract as constant.

File: src/main/java/mp4/Mp4Controller.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+			// for playListJson
+			// append the , for 2nd+ entries
+			if (!sbf.toString().endsWith("[")) {
+				sbf.append(",");
+			}
+			sbf.append("{\"url\":\"" + url + "\",\"fileName\":\"" + fileName + "\"}");
+		}
+		sbf.append("]}");
+
+		vMap.put("playListJson", sbf.toString());
```
Comment (domiyang) [inline src/main/java/mp4/Mp4Controller.java]: the key playListJson can be extract as constant.

File: src/main/java/mp4/Mp4Helper.java [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+		if (StringUtils.isEmpty(port)) {
+			port = "8080";
+			log.info("loaded with default port=" + port);
+		}
+		return Integer.parseInt(port);
+	}
+
+	/**
+	 * Get the accepted media file types, default to mp4,mov,mp3,m4v if not
+	 * provided from java arguments via D-mp4.media.types
```
Comment (domiyang) [inline src/main/java/mp4/Mp4Helper.java]: D-mp4.media.types => -Dmp4.media.types

File: src/main/java/mp4/Mp4Helper.java [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+		if (StringUtils.isEmpty(mediaTypes)) {
+			mediaTypes = "mp4,mov,mp3,m4v";
+			log.info("loaded with default videoSuffixList=" + mediaTypes);
+		}
+		return Arrays.asList(mediaTypes.split(","));
+	}
+
+	/**
+	 * Get the media root directory to stream, mandatory java arguments via
+	 * D-mp4.media.path
```
Comment (domiyang) [inline src/main/java/mp4/Mp4Helper.java]: D-mp4.media.path => -Dmp4.media.path

File: src/main/resources/webapp/index.html [hunk 9]
```diff
@@ -251,19 +208,17 @@ <h3>select video to play</h3>
...
 			</div>
-			<div class="panel-body">
-				<%=getListOfFilesLink()%>
-			</div>
+			<div class="panel-body" id="medialinks2"></div>
 		</div>
 	</div>
 	<footer class="footer">
-		<p>© 2016 dmb</p>
+		<p>@2018 dmb</p>
```
Comment (domiyang) [inline src/main/resources/webapp/index.html]: @ => ©

**Review by domiyang — CHANGES_REQUESTED** [review]
feedback with some line level comments, please check and make changes accordingly, thanks.

File: .settings/org.eclipse.wst.common.component (hunk 1 above)
Comment (yzx2107) [reply .settings/org.eclipse.wst.common.component]: done.

Comment (yzx2107) [reply .settings/org.eclipse.wst.common.component]: done with 022c320d16534f1c78695919d46b4180283a55d4

File: pom.xml (hunk 2 above)
Comment (yzx2107) [reply pom.xml]: done with 022c320d16534f1c78695919d46b4180283a55d4

File: src/main/java/mp4/Mp4Controller.java (hunk 3 above)
Comment (yzx2107) [reply src/main/java/mp4/Mp4Controller.java]: done with 022c320d16534f1c78695919d46b4180283a55d4

File: src/main/java/mp4/Mp4Controller.java (hunk 4 above)
Comment (yzx2107) [reply src/main/java/mp4/Mp4Controller.java]: done with 022c320d16534f1c78695919d46b4180283a55d4

File: src/main/java/mp4/Mp4Controller.java (hunk 5 above)
Comment (yzx2107) [reply src/main/java/mp4/Mp4Controller.java]: done with 022c320d16534f1c78695919d46b4180283a55d4

File: src/main/java/mp4/Mp4Helper.java (hunk 7 above)
Comment (yzx2107) [reply src/main/java/mp4/Mp4Helper.java]: done with 022c320d16534f1c78695919d46b4180283a55d4

File: src/main/java/mp4/Mp4Helper.java (hunk 8 above)
Comment (yzx2107) [reply src/main/java/mp4/Mp4Helper.java]: done with 022c320d16534f1c78695919d46b4180283a55d4

File: src/main/resources/webapp/index.html (hunk 9 above)
Comment (yzx2107) [reply src/main/resources/webapp/index.html]: done with 022c320d16534f1c78695919d46b4180283a55d4

File: src/main/java/mp4/Mp4Controller.java (hunk 6 above)
Comment (yzx2107) [reply src/main/java/mp4/Mp4Controller.java]: done with 022c320d16534f1c78695919d46b4180283a55d4

**Comment by yzx2107** [comment]
done with 022c320d16534f1c78695919d46b4180283a55d4
pls check again, thx.

**Review by domiyang — APPROVED** [review]
ok, good to go, thx for the effort.

**Pull request merged**